cvmfs/sync_item_dummy.h

```diff
@@ -10,9 +10,56 @@
 #include <ctime>
 #include <string>
 
+#include "ingestion/ingestion_source.h"
 #include "sync_union_tarball.h"
 
 namespace publish {
+
+class SyncItemDummyCatalog : public SyncItem {
+  friend class SyncUnionTarball;
+
+ protected:
+  SyncItemDummyCatalog(const std::string &relative_parent_path,
+                       const SyncUnion *union_engine)
+      : SyncItem(relative_parent_path, ".cvmfscatalog", union_engine,
+                 kItemFile) {}
+
+ public:
+  bool IsType(const SyncItemType expected_type) const {
+    return expected_type == kItemFile;
+  }
+
+  catalog::DirectoryEntryBase CreateBasicCatalogDirent() const {
+    catalog::DirectoryEntryBase dirent;
+    std::string name(".cvmfscatalog");
+    dirent.inode_ = catalog::DirectoryEntry::kInvalidInode;
+    dirent.linkcount_ = 1;
+    dirent.mode_ =
+        S_IFREG | S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH;
+    dirent.uid_ = getuid();
+    dirent.gid_ = getgid();
+    dirent.size_ = 0;
+    dirent.mtime_ = time(NULL);
+    dirent.checksum_ = this->GetContentHash();
+    dirent.is_external_file_ = this->IsExternalData();
```

Comment:
I think this should always be false (.cvmfscatalog files are not part of an external data set)

Reply:
Right!